tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs

```diff
@@ -82,7 +81,7 @@ protected override ITraceSampler GetSampler(ImmutableTracerSettings settings)
 
         protected override bool ShouldEnableRemoteConfiguration(ImmutableTracerSettings settings) => false;
 
-        protected override IAgentWriter GetAgentWriter(ImmutableTracerSettings settings, IDogStatsd statsd, Action<Dictionary<string, float>> updateSampleRates, IDiscoveryService discoveryService)
+        protected override IAgentWriter? GetAgentWriter(ImmutableTracerSettings settings, IDogStatsd statsd, Action<Dictionary<string, float>> updateSampleRates, IDiscoveryService discoveryService)
```

Comment:
I don't know if we _should_ make this return a nullable agentwriter? 🤔 Seeing as you're calling `Environment.FailFast` anyway, I think it would be fine to add a `null!`

Reply:
done in [3d98a45](https://github.com/DataDog/dd-trace-dotnet/pull/6374/commits/3d98a45a5bebdc0c7de9990b062d3e8eb506713d)